### ISSUE TITLE
Incorrectly identified recursive record in record scope

### ISSUE THREAD

**Issue by zaklogician** [agda/agda#5652]
**What happens:**

In a module, Agda accepts the following record and function definition without complaints:

```
record _suchThat_ (b : Set) (φ : Set → Set) : Set where
  field
    prop : φ b

compatible : Set → Set → Set
compatible x y = x suchThat (λ x' → y suchThat λ y' → y')
```

However, as soon as I try to use the exact same code *inside the scope of another `record` declaration*, Agda highlights `_suchThat_` in red, and reports the following error on the corresponding line:

> Recursive record `_suchThat_` needs to be declared as either inductive or coinductive

Removing the `compatible` function declaration resolves the error.

**What I expected to happen:**

I expected Agda to accept the record and function definition without complaints in both cases. I did not expect that the presence or absence of the `compatible` function declaration would affect the "recursive" status of the `_suchThat_` record. 

**Full code to reproduce:**

```
module PossibleBug where

record inside-some-record : Set where
  record _suchThat_ (b : Set) (φ : Set → Set) : Set where -- < -- error here
    field
      prop : φ b

  compatible : Set → Set → Set -- < -- removing this function resolves error
  compatible x y = x suchThat (λ x' → y suchThat λ y' → y')
```

**Version and environment:**

* Agda version 2.6.2, installed via cabal
* GNU Emacs 26.0
* Debian 10 (Linux kernel 5.13.6)

**Comment by nad** [comment]
You get the same problem if you replace `record inside-some-record : Set where` with `mutual`.

A similar problem: #2636.

**Comment by andreasabel** [comment]
Yeah, this is a known problem of the positivity checker.  A workaround it to switch it off with a suitable pragma. E.g. https://agda.readthedocs.io/en/v2.6.2/language/positivity-checking.html#the-no-positivity-check-pragma

Unfortunately, none of the developers wants to work on the positivity checker...  So this issue is likely to stay.

**Comment by andreasabel** [comment]
As @nad said, this is a known problem of the positivity check.  I recommend to work around it by following the complaint of Agda and declare the record as `inductive`.
```agda
record inside-some-record : Set where
  record _suchThat_ (b : Set) (φ : Set → Set) : Set where
    inductive  -- ADD ME!
    field
      prop : φ b

  compatible : Set → Set → Set
  compatible x y = x suchThat (λ x' → y suchThat λ y' → y')
```
